### PR TITLE
VER-81658: Fix prepared statement tests

### PR DESCRIPTION
**Pull request by jonathanl-bq**
This PR fixes parameterized queries by treating them the same way that other drivers (e.g. JDBC, ADO.NET) treat anonymous prepared statements. In the other client drivers, these prepared statements have a name generated for them, "s" + statement counter.

In addition, the PR fixes all documented failing tests that used parameterized queries or prepared statements.

### CODE REVIEW

File: packages/v-pool/test/connection-timeout.js [hunk 1]
```diff
@@ -128,7 +128,7 @@ describe('connection timeout', () => {
         expect(err).to.be.an(Error)
         expect(result).to.be(undefined)
         release()
-        pool.query('select $1::text as name', ['brianc'], (err, res) => {
+        pool.query('select ?::varchar as name', ['brianc'], (err, res) => {
```
Comment (williammatherjones) [inline packages/v-pool/test/connection-timeout.js:131]: Think about the best way to keep track of the fact we had to change params from $1 to ? style.   Should we keep track of  issues like this in the postrgres / vertica protocol confluence pages?   This is another category of change that needs to be made for code designed to work with Postgres to work with Vertica and the only fix would be for vertica to understand $1 notation.

File: packages/vertica-nodejs/lib/connection.js [hunk 2]
```diff
@@ -22,6 +22,10 @@ class Connection extends EventEmitter {
     this.ssl = config.ssl || false
     this._ending = false
     this._emitMessage = false
+    this.statementCounterBuffer = new SharedArrayBuffer(32)
```
Comment (williammatherjones) [inline packages/vertica-nodejs/lib/connection.js:25]: Why was this needed?  I don't see where it is used?

File: packages/vertica-nodejs/lib/connection.js [hunk 3]
```diff
@@ -216,6 +220,10 @@ class Connection extends EventEmitter {
   sendCopyFail(msg) {
     this._send(serialize.copyFail(msg))
   }
+
```
Comment (williammatherjones) [inline packages/vertica-nodejs/lib/connection.js:223]: Where is this called from?

File: packages/vertica-nodejs/lib/query.js [hunk 4]
```diff
@@ -190,6 +190,8 @@ class Query extends EventEmitter {
     // complete or when an error is encountered
     this.isPreparedStatement = true
 
+    this.name = this.name || connection.makeStatementName()
```
Comment (williammatherjones) [inline packages/vertica-nodejs/lib/query.js:193]: Remind me what this does in Javascript?

**Review by williammatherjones — APPROVED** [review]

File: packages/vertica-nodejs/lib/connection.js (hunk 2 above)
Comment (jonathanl-bq) [reply packages/vertica-nodejs/lib/connection.js:25]: JavaScript doesn't seem to have AtomicIntegers like Java, so I had to use this Int32 Array instead. JDBC uses an AtomicInteger m_statementCount to generate names for unnamed prepared statements.

File: packages/vertica-nodejs/lib/query.js (hunk 4 above)
Comment (jonathanl-bq) [reply packages/vertica-nodejs/lib/query.js:193]: It checks if this.name is null/undefined. If it is, then it defaults to using the value on the right side of the ||. Otherwise, it just uses the this.name value.

File: packages/v-pool/test/connection-timeout.js (hunk 1 above)
Comment (jonathanl-bq) [reply packages/v-pool/test/connection-timeout.js:131]: I added it to this Confluence page "Differences Between Node Postgres and Vertica-NodeJS Drivers". I think that should be sufficient for tracking these kinds of changes, but I'm open to other ideas too.

**Review by jeremyp-bq — APPROVED** [review]

**Review by DMickens — APPROVED** [review]
Looks good to me

File: packages/vertica-nodejs/lib/query.js [hunk 5]
```diff
@@ -197,6 +201,15 @@ class Query extends EventEmitter {
       })
     }
 
+    connection.describe({
```
[the remark this answers is not in the document]
Comment (jonathanl-bq) [reply packages/vertica-nodejs/lib/query.js:204]: Yes, it's not a static variable. It is initialized to 0 for each new Connection.
